R/mobr.R

```diff
@@ -214,32 +212,39 @@ rarefaction = function(x, method, effort=NULL) {
     }
     out = rowSums(1 - p)
     names(out) = effort
-    class(out) = 'rarefaction'
+    #class(out) = 'rarefaction'
     return(out)
 }
 
-# Auxillary function: difference between the ind-based rarefaction and the sample-based rarefaction for one group
-#   with the evaluation sample size (number of individuals) defined by ref_dens,
-#   evaluated at specified points (given by "effort")
-# New input: swap - if swap is FALSE, S_samp_rare is generated on the input comm_group
-#   if swap is TRUE, S_swap_rare is generated by Dan's new function avg_swap_rare()
-# Output: a two-column data frame, with sample size (effort)
-#   and deltaS (effect of N)
-effect_of_N = function(comm_group, ref_dens, effort, swap = FALSE){
-  group_sad = colSums(comm_group)
-  if (swap == FALSE)
-    S_samp_rare = as.numeric(rarefaction(comm_group, 'samp'))
-  else
-    S_samp_rare = avg_swap_rare(comm_group, rep(1, nrow(comm_group)), nperm = 100)
-  effort = effort[which(effort <= min(sum(comm_group), ref_dens * nrow(comm_group)))]
-  interp_S_samp_rare = pchip(c(1, ref_dens * (1:nrow(comm_group))), c(1, S_samp_rare), effort)
-  S_indiv_rare = rarefaction(group_sad, 'indiv', effort = effort)
-  deltaS = interp_S_samp_rare - as.numeric(S_indiv_rare)
-  out = data.frame(cbind(effort, deltaS))
-  names(out) = c('effort', 'deltaS')
+# Auxillary function: difference between the ind-based rarefaction and the
+# sample-based rarefaction for one group with the evaluation sample size (number
+# of individuals) defined by ref_dens, evaluated at specified points (given by
+# "effort") New input: permute - if permute is FALSE, S_samp_rare is generated
+# on the input comm_group if permute is TRUE, S_swap_rare is generated by Dan's
+# new function avg_swap_rare() Output: a two-column data frame, with sample size
+# (effort) and deltaS (effect of N)
+effect_of_N = function(comm_group, ref_dens, inds){
+  nplots = nrow(comm_group)
+  group_dens = sum(comm_group) / nplots
+  # calcualte the number of individuals sampled
+  # as each of the nplots are collected
+  samp_effort = round((1:nplots) * group_dens)
+  #effort = effort[which(effort <= min(sum(comm_group), ref_dens * nrow(comm_group)))]
+  # use individual based rarefaction evaluated at the 
+  # number of individuals sampled for each plot
+  S_samp = rarefaction(comm_group, 'indiv', samp_effort)
+  # rescale and interpolate this plot based S to individual based 
+  # using the ref_density (i.e., not the observed density)
+  rescaled_effort = round(1:nplots * ref_dens)
+  interp_S_samp = pchip(c(1, rescaled_effort),
```

Comment:
Would there be any problem if `max(inds)` is larger than `max(rescaled_effort)` (ie extrapolation)?